### PR TITLE
Save extra comment fields

### PR DESCRIPTION
**Pull request by vrajmohan**

### CODE REVIEW

**Comment by vrajmohan** [comment]
Fixes https://github.com/eregs/notice-and-comment/issues/115

File: regulations/docket.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    return fields
+
+
+def sanitize_fields(body):
+    document_fields = get_document_fields(settings.COMMENT_DOCUMENT_ID)
+    for name, field in document_fields.iteritems():
+        if field.required and name not in body:
+            return False, "Field {} is required".format(name)
+        if name in body:
+            if len(body[name]) > field.max_length:
```
Comment (cmc333333) [inline regulations/docket.py]: Very minor, but could these two `ifs` be combined with an `and` to reduce the depth?

Comment (cmc333333) [inline regulations/docket.py]: Anything funky around unicode we should worry about here? Just thinking aloud.

File: regulations/docket.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                return False, "Field {} exceeds expected length of {}".format(
+                    name, field.max_length)
+
+    # Remove extra fields if any
+    extra_fields = [field for field in body if field not in document_fields]
+    for field in extra_fields:
+        del body[field]
```
Comment (cmc333333) [inline regulations/docket.py]: Good idea!

**Comment by cmc333333** [comment]
Would you mind adding tests around `sanitize_fields`? Unrelated, I'm a tad concerned that we're writing our own validation framework -- would it make sense to reuse Django's or find a more general purpose version in a later step?

File: regulations/docket.py (hunk 1 above)
Comment (vrajmohan) [inline regulations/docket.py]: Good point. We didn't talk about or see any documentation around character encodings that regulations.gov accepts. Let me poke around.

**Comment by vrajmohan** [comment]
I'll add tests, and not just for `sanitize_fields` - I wanted to make sure that we are OK with my approach.
Regarding the validation framework, I think Django's is tightly coupled with their models. Yes, it makes sense to find a more general purpose library if we are going to do much more of this.

File: regulations/tests/sanitize_fields_tests.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        self.assertEqual("Field required_field is required", message)
+
+    def test_extra_field(self):
+        test_body = {
+            "required_field": "Value 1",
+            "extra_field": "Value 2"
+        }
+        valid, message = sanitize_fields(test_body)
+        self.assertTrue(valid)
+        self.assertTrue("extra_field removed", "extra_field" not in test_body)
```
Comment (cmc333333) [inline regulations/tests/sanitize_fields_tests.py]: I think these two params are reversed

Comment (vrajmohan) [inline regulations/tests/sanitize_fields_tests.py]: Good catch. Damn JUnit.

**Comment by cmc333333** [comment]
Good stuff

**Pull request merged**